srtcore/sync.cpp

```diff
@@ -527,3 +535,80 @@ void srt::sync::CTimer::tick()
     m_event.notify_one();
 }
 
+
+void srt::sync::CGlobEvent::triggerEvent()
+{
+    return g_Sync.notify_one();
+}
+
+bool srt::sync::CGlobEvent::waitForEvent()
+{
+    return g_Sync.lock_wait_for(milliseconds_from(10));
+}
+
+////////////////////////////////////////////////////////////////////////////////
+//
+// CThread class
+//
+////////////////////////////////////////////////////////////////////////////////
+#ifndef USE_STDCXX_CHRONO
+
+srt::sync::CThread::CThread()
+{
+    m_thread = pthread_t();
+}
+
+srt::sync::CThread::CThread(void *(*start_routine) (void *), void *arg)
+{
+    const int st = pthread_create(&m_thread, NULL, start_routine, arg);
+    if (st != 0)
+    {
+        LOGC(mglog.Error, log << "pthread_create failed with " << st);
+        //throw std::system_error()
+    }
+}
+
+#if HAVE_FULL_CXX11
+srt::sync::CThread& srt::sync::CThread::operator=(CThread&& other)
+#else
+srt::sync::CThread& srt::sync::CThread::operator=(CThread& other)
+#endif
+{
+    if (joinable())
```

Comment:
Well, not sure if this follows the semantics of C++11. Shouldn't an assignment over an existing thread cause exception?

Reply:
C++11 calls `terminate` if the thread being assigned was not terminated.
Reference: [link](https://en.cppreference.com/w/cpp/thread/thread/operator%3D).

Reply:
If you want that `CThread` be an alias to `std::thread`, then `CThread` must follow its semantics. Even if it means calling `terminate()`.

Reply:
Added `pthread_cancel(m_thread)` for release builds.